### DISCUSSION TITLE
x-on:click not working without x-data

### DISCUSSION THREAD

**Discussion by michabbb** [alpinejs/alpine#3036] [1. Help]
hi,
i am using alpinejs within laravel jetstream (with livewire).
the documentation says:

https://alpinejs.dev/directives/on

this should work:

```html
<button x-on:click="alert('Hello World!')">Say Hi</button>
```
but actually, for me the only way to make this trigger work, is doing it like this:

```html
<div x-data="{}">
    <button x-on:click="alert('Hello World!')">Say Hi</button>
</div>
```

so I am wondering, does this has anything to do with livewire or what could be the reason here ??

thanks for any help!

**Comment by SimoTod** [comment]
It's expected and this is the way alpine work, you ccan read the x-data section for more details. Directives only work inside a component. That example is just an extract of a component to focus on the feature. Note, an empty x-data can be put on the tag itself if you don't need anything else
`<button x-data x-on:click="...`

**Reply by michabbb** [reply]
thanks, I will test. but I have to say, showing only 50% of something in the documentation is a very bad practice, because people who are trying to understand usually tend to copy&paste stuff to test what they see, in their own code. so showing not working pieces of code is very confusing and misleading. anyway, thanks for the clarification here.
